utilities/src/Types/Transaction.res

```diff
@@ -0,0 +1,145 @@
+  | #view
+  | #component
+  | #task
+]
+type dynamicSamplingContext
+type transactionMetadata = {
+  /** The sample rate used when sampling this transaction */
+  sampleRate?: float,
+  /**
+   * The Dynamic Sampling Context of a transaction. If provided during transaction creation, its Dynamic Sampling
+   * Context Will be frozen
+   */
+  dynamicSamplingContext?: option<dynamicSamplingContext>,
+  /** For transactions tracing server-side request handling, the request being tracked. */
+  request?: Polymorphics.polymorphicRequest,
+  /** Compatibility shim for transitioning to the `RequestData` integration. The options passed to our Express request
+   * handler controlling what request data is added to the event.
+   * TODO (v8): This should go away
+   */
+  requestDataOptionsFromExpressHandler?: Js.Dict.t<unknown>,
+  /** For transactions tracing server-side request handling, the path of the request being tracked. */
+  /** TODO: If we rm -rf `instrumentServer`, this can go, too */
+  requestPath?: string,
+  /** Information on how a transaction name was generated. */
+  source: transactionSource,
+  /** Metadata for the transaction's spans, keyed by spanId */
+  spanMetadata: Js.Json.t,
+}
+
+/**
+ * Context data passed by the user when starting a transaction, to be used by the tracesSampler method.
+ */
+type customSamplingContext = {
+    _additionalTypes: Js.Json.t
+}
+/**
+ * Data pulled from a `sentry-trace` header
+ */
+type traceparentData = Js.Json.t
+
+type transactionContext = {
+  ...Span.spanContext,
+  /**
+   * If true, sets the end timestamp of the transaction to the highest timestamp of child spans, trimming
+   * the duration of the transaction. This is useful to discard extra time in the transaction that is not
+   * accounted for in child spans, like what happens in the idle transaction Tracing integration, where we finish the
+   * transaction after a given "idle time" and we don't want this "idle time" to be part of the transaction.
+   */
+  trimEnd?: bool,
+  /**
+   * If this transaction has a parent, the parent's sampling decision
+   */
+  parentSampled?: bool,
+  /**
+   * Metadata associated with the transaction, for internal SDK use.
+   */
+  metadata?: option<transactionMetadata>,
+}
+
+type samplingContext = {
+  ...customSamplingContext,
+  /**
+   * Context data with which transaction being sampled was created
+   */
+  transactionContext: transactionContext,
+  /**
+   * Sampling decision from the parent transaction, if any.
+   */
+  parentSampled?: bool,
+  /**
+   * Object representing the URL of the current page or worker script. Passed by default when using the `BrowserTracing`
+   * integration.
+   */
+  location?: Misc.workerLocation,
+  /**
+   * Object representing the incoming request to a node server. Passed by default when using the TracingHandler.
+   */
+  request?: Misc.extractedNodeRequestData,
+}
+
+type transaction = {
+  ///TODO figure out how to exculde type while type spreading
```

Comment:
Add those types which are needed manually and make everything optional by adding `?`